server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -584,7 +584,12 @@ public Operation readOperation(Location location) throws IOException {
             if (current.generation == location.generation) {
                 // no need to fsync here the read operation will ensure that buffers are written to disk
                 // if they are still in RAM and we are reading onto that position
-                return current.read(location);
+                try {
+                    return current.read(location);
+                } catch (final IOException e) {
+                    closeOnTragicEvent(e);
+                    throw e;
```

Comment:
The other calls to `closeOnTragicEvent` are all of the form
```
try {
    closeOnTragicEvent(ex);
} catch (final Exception inner) {
    ex.addSuppressed(inner);
}
throw ex;
```

which does not make any sense btw when you look at `closeOnTragicEvent` which already takes care of exceptions that happen during closing. Can you unify the code in this class?

Reply:
Okay, I would like to do this in an immediate follow-up to this one. For now I pushed: f9a88eb6216155c6b7413fc59d5e0cb18e49aab3